src/mongocrypt.c

```diff
@@ -1518,7 +1559,7 @@ _mongocrypt_parse_kms_providers (
       }
    }
 
-   if (log->trace_enabled) {
+   if (log && log->trace_enabled) {
```

Comment:
I actually considered that, but it seemed less intrusive to go this route.  Your top-level comment included this bit:

> There are some cases where a BSON_ASSERT or return false is applied to non-essential features (i.e. logging). These should be maintained as non-obtrusive optional behavior rather than treated as error conditions.

Do I take this to mean that in this case (where the `log` variable is reasonable expected to never be `NULL`) that the assertion is the right way, but that in other cases (say perhaps where the user might be able to call the function in question and thus potentially pass a `NULL` intentionally) then something like `log &&` would be better?

Reply:
I went ahead an made the suggested change here.  I went with 

```
   if (!log) {
      CLIENT_ERR ("argument 'log' is required");
      return false;
   }
```
at the top of the function.  It will show up in the next commit.

Reply:
You make a good point that your initially proposed change of `if (log && ...)` would be more consistent with my suggestion elsewhere about non-essential features such as logging being considered a non-error. I think I may prefer the `log &&` approach over an assertion, but I also think either would be preferable to a `if (!log) { CLIENT_ERR(...); return false; }`.